internal/k8s/controller.go

```diff
@@ -976,8 +993,58 @@ func (lbc *LoadBalancerController) syncVirtualServer(task task) {
 				glog.Errorf("Error when updating the status for VirtualServerRoute %v/%v: %v", vsr.Namespace, vsr.Name, err)
 			}
 		}
+
+		vsrkey := fmt.Sprintf("%s/%s", vsr.Namespace, vsr.Name)
+		vsrlist = checkForVirtualServerRoute(vsrkey, vsrlist)
+	}
+	reason := "Ignored"
+	for _, vsr := range vsrlist {
+		msg := fmt.Sprintf("Ignored by VirtualServer %v/%v", vs.Namespace, vs.Name)
+		lbc.recorder.Eventf(vsr, api_v1.EventTypeWarning, "Ignored", msg)
+		err := lbc.statusUpdater.UpdateVirtualServerRouteStatus(vsr, conf_v1.StateInvalid, reason, msg)
+		if err != nil {
+			glog.Errorf("Error when updating the status for VirtualServerRoute %v/%v: %v", vsr.Namespace, vsr.Name, err)
+		}
+	}
+}
+
+func checkForVirtualServerRoute(vsrkey string, vsrlist []*conf_v1.VirtualServerRoute) []*conf_v1.VirtualServerRoute {
+	var vsrNotRef []*conf_v1.VirtualServerRoute
```

Comment:
I suggest removing this slice and modifying the slice from args directly.

The return value is a slice of vsrs, not a slice of vsrsNotReferenced.